tests/test_atomicio.py

```
@@ -0,0 +1,287 @@
+    assert target.read_text() == "new"
+    assert len(calls) == 3
+
+
+def test_a_permanently_locked_destination_raises_and_leaves_no_debris(tmp_path):
+    """Retrying forever would hang the save; giving up must still leave the
+    old file intact and no .tmp behind."""
+    target = tmp_path / "out.json"
+    target.write_text("old")
+    real = atomicio.os.replace
+
+    def always(src, dst):
```

Ruff (0.16.2)

[warning] Missing return type annotation for private function `always`

Add return type annotation: `Never`

(ANN202)